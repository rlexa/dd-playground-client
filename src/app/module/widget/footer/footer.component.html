<app-version></app-version>
<ng-content></ng-content>
