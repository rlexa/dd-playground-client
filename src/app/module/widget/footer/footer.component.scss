:host {
  display: flex;
  align-items: center;
  padding: var(--size-gap-narrow);
  border-top: solid 1px var(--color-border);

  .filler {
    flex: 1 1 auto;
  }
}
