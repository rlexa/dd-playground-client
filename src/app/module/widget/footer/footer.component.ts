import {CommonModule} from '@angular/common';
import {ChangeDetectionStrategy, Component} from '@angular/core';
import {VersionComponent} from '../version';

@Component({
  selector: 'app-footer',
  template: `<span class="filler"></span>
    <ng-content />
    <app-version />`,
  styleUrls: ['./footer.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  standalone: true,
  imports: [CommonModule, VersionComponent],
})
export class FooterComponent {}
