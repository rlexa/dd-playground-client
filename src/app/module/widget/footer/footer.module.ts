import {CommonModule} from '@angular/common';
import {NgModule} from '@angular/core';
import {VersionModule} from '../version';
import {FooterComponent} from './footer.component';

@NgModule({imports: [CommonModule, VersionModule], exports: [FooterComponent], declarations: [FooterComponent]})
class FooterModule {}

export {FooterModule, FooterComponent};
