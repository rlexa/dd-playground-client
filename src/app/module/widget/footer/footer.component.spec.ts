import {ComponentFixture} from '@angular/core/testing';
import {MockBuilder, MockRender} from 'ng-mocks';
import {FooterComponent} from './footer.component';

describe('FooterComponent', () => {
  let fixture: ComponentFixture<FooterComponent>;

  beforeEach(() => MockBuilder(FooterComponent));

  beforeEach(() => {
    fixture = MockRender(FooterComponent);
    fixture.detectChanges();
  });

  test('renders', () => expect(fixture).toMatchSnapshot());
});
