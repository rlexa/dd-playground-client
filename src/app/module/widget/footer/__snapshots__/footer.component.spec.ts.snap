// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent renders 1`] = `
<app-footer>
  <span
    class="filler"
  /><app-version />
</app-footer>
`;
